python/lsst/afw/cameraGeom/camera.py

```diff
@@ -106,8 +129,38 @@ def findDetectors(self, cameraPoint):
                 detectorList.append(detector)
         return detectorList
 
+    def findDetectorsArray(self, xCoord, yCoord, coordSys):
+        """!Find the detectors that cover a list of points specified by x and y coordinates in any system
+
+        @param[in] xCoord is an array of x coordinates
+        @param[in] yCoord is an array of y coordinates
+        @param[in] coordSys is the coordinate system in which xCoord and yCoord are defined
+        @return a list of lists; each list contains the names of all detectors which contain the
+        corresponding point
+        """
+
+        cameraPoints = [afwGeom.Point2D(x,y) for x,y in zip(xCoord, yCoord)]
+
+        #transform the points to the native coordinate system
+        nativePoints = self._transformSingleSysArray(cameraPoints, coordSys, self._nativeCameraSys)
+
+        detectorList = []
+        for i in range(len(xCoord)):
+            detectorList.append([])
+
+        for detector in self:
+            coordMap = detector.getTransformMap()
+            cameraSys = detector.makeCameraSys(PIXELS)
+            detectorPoints = coordMap.transform(nativePoints, self._nativeCameraSys, cameraSys)
+            box = afwGeom.Box2D(detector.getBBox())
+            for i, pt in enumerate(detectorPoints):
+                if box.contains(pt):
+                    detectorList[i].append(detector)
+
+        return detectorList
+
     def getTransformMap(self):
-        """!Obtain a pointer to the transform registry.  
+        """!Obtain a pointer to the transform registry.
```

Comment:
While you are fixing things, could you please get rid of "a pointer to"? Pointers are a C++ concept.